flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -1271,6 +1276,8 @@ public InlineElement getDescription() {
      * @param classLoader a class loader to use when loading classes
      */
     public void configure(ReadableConfig configuration, ClassLoader classLoader) {
+        configuration.toMap().forEach(this.configuration::setString);
+
```

Comment:
This allows `ExecutionConfig`  to pick up the unknown options without being aware of them.

The downsides are:
- conversion to string (and from string when reading)
- picking up everything

However, I don't see any negative impact.

cc: @twalthr , @pnowojski, @StefanRRichter, @zentol

Reply:
How about we simply add a `set(ConfigOption)` and `set(String)` to `ExecutionConfig`? It is not really necessary that ExecutionConfig picks up all configuration from `ReadableConfig configuration`. For a later runtime patching of ExecutionConfig, we could only use the `set(String)`.